Sources/MarathonCore/Script.swift

```diff
@@ -131,6 +136,35 @@ internal final class Script {
                 printer.output("✏️  Opening \(relativePath)")
 
                 try shellOut(to: "open \"\(path)\"", printer: printer)
+                
+                // This looks bad, but hear me out. First, read the two comments below about opening Xcode to a specific file.
+                // The reason for the sleep is that if you shell out two `open` commands back to back, if Xcode hasn't finished 
+                // opening the first file (which can take upwards of a second) before the second `open` is executed, it will
+                // open the second file in a new window. By sleeping for a second, we give Xcode a reasonable chance to finish opening
+                // the first file before trying to open the second. By sleeping at this point, the user shouldn't notice a difference
+                // except for the "i" message below about commits, since the first open has already been executed.
+                //
+                // Other approaches I've tried for avoiding potentially opening a second window:
+                // 1. Edit the .xcuserstate before opening Xcode to spoof the window state and force Xcode to open to a specific file
+                //
+                // I've asked a question on StackOverflow to try find a better way to do this:
+                // https://stackoverflow.com/questions/44346442/open-xcode-project-and-set-active-file-from-terminal
+                sleep(1)
```

Comment:
This is clever! 😄 I've been experimenting with different ways to get Xcode to open to a specific file, both for Marathon (would be super nice if it opened at the script file directly), and for tools like TestDrive. I never considered that Xcode actually auto-opens the file if a project containing it is already open, so this is super awesome 🚀 I'm thinking, instead of sleeping, how about we observe whether Xcode is opened? Should be a way to do that using the system APIs.